packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapController.java

```diff
@@ -508,6 +508,10 @@ public void dispose() {
   }
 
   private void setGoogleMapListener(@Nullable GoogleMapListener listener) {
+    if (googleMap == null) {
+      Log.v(TAG, "Controller was disposed before GoogleMap was ready.");
```

Comment:
It's only a dispose if `listener` is null, right? It seems like we should either be checking for that (if that's the only time we expect `googleMap` to be null), or making this error message conditional or generic.

Reply:
It's possible that `listener` can be `null` while `googleMap` is non-null. This is done in `dispose` to pass `null` to all the `googleMap.set....Listener()` methods to prevent accidental callbacks to Dart.

Also, this class implements `GoogleMapListener`, so the class is either passing itself or null.

This conditional will also only trigger in the rare scenario where `onMapReady`, which is the only place `googleMap` is set, isn't called before `dispose`. I couldn't reproduce this scenario locally, but it happens on FTL for some reason.

Reply:
Oh, I see; this is handling a post-dispose case, not a call from `dispose`.